packages/web/src/components/play-bar/playback-rate-button/PlaybackRateButton.tsx

```diff
@@ -0,0 +1,81 @@
+import { PlaybackRate, playerActions, playerSelectors } from '@audius/common'
+import { PopupMenu, PopupMenuItem, PopupPosition } from '@audius/stems'
+import cn from 'classnames'
+import { useDispatch, useSelector } from 'react-redux'
+
+import { ReactComponent as Icon0_5x } from 'assets/img/iconPlaybackRate0_5x.svg'
+import { ReactComponent as Icon0_8x } from 'assets/img/iconPlaybackRate0_8x.svg'
+import { ReactComponent as Icon1_1x } from 'assets/img/iconPlaybackRate1_1x.svg'
+import { ReactComponent as Icon1_2x } from 'assets/img/iconPlaybackRate1_2x.svg'
+import { ReactComponent as Icon1_5x } from 'assets/img/iconPlaybackRate1_5x.svg'
+import { ReactComponent as Icon1x } from 'assets/img/iconPlaybackRate1x.svg'
+import { ReactComponent as Icon2_5x } from 'assets/img/iconPlaybackRate2_5x.svg'
+import { ReactComponent as Icon2x } from 'assets/img/iconPlaybackRate2x.svg'
+import { ReactComponent as Icon3x } from 'assets/img/iconPlaybackRate3x.svg'
+import zIndex from 'utils/zIndex'
+
+import styles from '../PlayBarButton.module.css'
+
+const { setPlaybackRate } = playerActions
+const { getPlaybackRate } = playerSelectors
+
+export type PlaybackRateButtonProps = {
+  isMobile: boolean
+}
+
+const playbackRateIconMap: Record<PlaybackRate, typeof Icon1x> = {
+  '0.5x': Icon0_5x,
+  '0.8x': Icon0_8x,
+  '1x': Icon1x,
+  '1.1x': Icon1_1x,
+  '1.2x': Icon1_2x,
+  '1.5x': Icon1_5x,
+  '2x': Icon2x,
+  '2.5x': Icon2_5x,
+  '3x': Icon3x
+}
+
+export const PlaybackRateButton = ({ isMobile }: PlaybackRateButtonProps) => {
+  const dispatch = useDispatch()
+  const setRate = (rate: PlaybackRate) => dispatch(setPlaybackRate({ rate }))
+  const playbackRate = useSelector(getPlaybackRate)
+  const Icon = playbackRateIconMap[playbackRate]
+  const rates: PlaybackRate[] = [
+    '0.5x',
+    '0.8x',
+    '1x',
+    '1.1x',
+    '1.2x',
+    '1.5x',
+    '2x',
+    '2.5x',
+    '3x'
+  ]
+
+  const items: PopupMenuItem[] = rates.map((rate) => ({
```

Comment:
it might make sense to memoize this. would require a bit of changes with `setRate` too